Docs work typography

tkoleary opened the pull request:
Changed:
  - project: Core
    component: Storybook
    description:  Updates to typography storybook file as well as styles and theme in storybook
    issue: 
    impact: Patch

tkoleary committed:
Imported mayflower theme to docs

For some reason storybook requires it to be impoerted there as well to available in the docs pages.
Also made somee tweaks to the Mayflower sstorybook theme.

tkoleary committed:
Updates to styles, markup and content for typography.

Plus a few minor content changes to 'about'

clairesunstudio commented on packages/core/stories/tokens/typography/Typography.stories.mdx:
```diff
@@ -1,7 +1,7 @@
 import DecorativeLink from '@massds/mayflower-react/dist/DecorativeLink';
 import Heading from '@massds/mayflower-react/dist/Heading';
-// import RichText from '@massds/mayflower-react/dist/RichText';
+// import p  from '@massds/mayflower-react/dist/p ';
```
```suggestion
```

clairesunstudio commented on packages/core/stories/about/Usage.stories.mdx:
```diff
@@ -103,13 +107,15 @@ For examples:
-To import fonts and get started with basic Mayflower layout CSS, you need to import `scss/global.scss` (@use `scss/global.scss`) or link to the compiled `static/fonts/css/global.css` (a minified version is available as global.min.css).
-To implement a Mayflower branding component in your website, navigate to the documentation of the component and copy and paste the html and CSS (and JS) snippets into your code to get started. You can also link to the CDN version of CSS and JS files for the component directly in the head of your HTML.
+
+To import fonts and get started with basic Mayflower layout CSS, import `scss/global.scss` (`@use scss/global.scss`) or link to the compiled `static/fonts/css/global.css` (or the minified version, global.min.css). To use a Mayflower branding component on your site, go to the documentation of the component and copy and paste the html and link to the CDN version of CSS and JS files in the head of y ...
```
```suggestion
To import fonts and get started with basic Mayflower layout CSS, import `scss/global.scss` (`@use scss/global.scss`) or link to the compiled `static/fonts/css/global.css` (or the minified version, global.min.css). To use a Mayflower branding component on your site, go to the documentation of the component and copy and paste the html and link to the CDN version of CSS and JS files in the head of your HTML. You can also host the CSS and JS downloaded from mayflower assets package.
```

We want to provide the mayflower core users the convenience to get everything they need in this documentation, without them having to delve into the monorepo. I think we should ask them to link to or download from the CDN provided in the code snippet, or for more advanced users who uses npm, to link the file from their locally installed mayflower-assets package

clairesunstudio commented on packages/core/stories/tokens/typography/Typography.stories.mdx:
```diff
@@ -23,146 +24,182 @@ import generateTitle from '../../util/generateTitle';
-- To support the guiding principles of the Mass.gov brand identities: modern, dignified, human, helpful
+# Typography
+
+———
```
```suggestion
```

clairesunstudio commented on packages/core/.storybook/preview.js:
```diff
@@ -16,5 +18,8 @@ export const parameters = {
-  viewMode: 'docs'
+  viewMode: 'docs',
+	docs: {
+    theme: mayflowerTheme,
```
nice catch! 

Interesting, I was referencing their config doc and it says to import in the manager.js instead of preview.js
https://storybook.js.org/docs/react/configure/theming

Where did you find this?

tkoleary replied:
Thanks!  Got it here: https://storybook.js.org/docs/react/configure/theming#theming-docs

clairesunstudio replied:
huh thanks! I was looking at the global config part

clairesunstudio commented on packages/core/.storybook/mayflowerTheme.js:
```diff
@@ -3,17 +3,17 @@ import logo from './mayflower-logo.png';
 
   // Typography
-  fontBase: '"Noto Sans VF", "Noto Sans", "Helvetica", "Arial", sans-serif',
+  fontBase: '"Noto Sans VF", "Noto Sans", sans-serif',
```
can we put back the two fallback fonts, to be consistent with mayflower font settings?

tkoleary replied:
Yes, but not those two, they don't match stylistically. I will follow up with some that do.

tkoleary replied:
Probably should patch the main repo too.

clairesunstudio replied:
I think the purpose here is to control the fallback fonts, when user turns off webfonts in their browsers. They'll need to be popular system fonts

tkoleary committed:
Update packages/core/stories/tokens/typography/Typography.stories.mdx

Co-authored-by: Minghua Sun <clairesunstudio@gmail.com>

tkoleary committed:
Update packages/core/stories/tokens/typography/Typography.stories.mdx

Co-authored-by: Minghua Sun <clairesunstudio@gmail.com>

tkoleary committed:
Update packages/core/stories/about/Usage.stories.mdx

Co-authored-by: Minghua Sun <clairesunstudio@gmail.com>

clairesunstudio commented on packages/core/stories/tokens/typography/Typography.stories.mdx:
```diff
@@ -23,146 +23,181 @@ import generateTitle from '../../util/generateTitle';
-</RichText>
 
+<div>
+	<p className="font-specimen large">{letters.toUpperCase()}<br />{letters}</p>
```
I think we should also display the numeric digits here. Is there a reason those are removed?

tkoleary replied:
Good point. I thought it was redundant since they are in the block of all characters, but it is nice to have them with the alphabet.

tkoleary replied:
I put them back.

clairesunstudio commented:
Removing the outline styles makes the sections a little hard to read for me. 
![Screen Shot 2020-11-23 at 5 39 40 PM](https://user-images.githubusercontent.com/5789411/100023461-061ea680-2db3-11eb-9f01-71b6ec4a09d2.png)



Because of the font sizes variations, the hierarchy can be confusing 
Can we style the language section something like this:
![Screen Shot 2020-11-23 at 5 27 07 PM](https://user-images.githubusercontent.com/5789411/100023335-ca83dc80-2db2-11eb-9b62-89664cb3ab11.png)

clairesunstudio commented:
Same for these sections, it's hard for the eyes to separate out the specimen vs the actual headings. I think will help to add back the `"sb-block"` class. 
![Screen Shot 2020-11-23 at 5 28 41 PM](https://user-images.githubusercontent.com/5789411/100023571-349c8180-2db3-11eb-89ae-e115c3069012.png)

The weights and the class name variants may look better in a table. 

It'd be easier for me to adjust and comment on these by editing the code directly. Let me know if you want me to do that

tkoleary committed:
Updated vertical spacing

-reverted fallbacks
-reverted numbers
-added class for language labels
-added language labels to languages that didn't have them
-indented language labels to differentiate from scripts
-Corrected script names
-added rtl class for labels
-Moved sample to top

tkoleary commented:
I think my latest push resolves the issues you pointed out. Let me know if there's anything else you see.

clairesunstudio commented on packages/core/stories/about/Usage.stories.mdx:
```diff
@@ -3,9 +3,12 @@ import { Meta } from '@storybook/addon-docs/blocks';
 import generateTitle from '../util/generateTitle';
 const { STORYBOOK_CDN_PATH, STORYBOOK_CDN, STORYBOOK_PKG } = process.env;
 
+import '../styles/index.scss';
```
```suggestion
```

Storybook only needs to import scss once

clairesunstudio commented on packages/core/stories/tokens/typography/Typography.stories.mdx:
```diff
@@ -15,6 +14,7 @@ const slugifyStyle = (style) => {
   return italic ? `${weight.toLowerCase()}--${italic.toLowerCase()}` : weight.toLowerCase();
 };
 import generateTitle from '../../util/generateTitle';
+import '../../styles/index.scss';
```
```suggestion
```

clairesunstudio committed:
Apply suggestions from code review
